demos/platforms/lpc4078.cpp

```diff
@@ -42,12 +43,22 @@ resource_list initialize_platform()
                                   .baud_rate = 115200,
                                 });
 
+  // Initialize LED pin
   static hal::lpc40::output_pin led(1, 10);
 
+  // Get and initialize SPI2 for SPI communication
+  static hal::lpc40::spi spi2(2,
+                              hal::lpc40::spi::settings{
+                                4.0_MHz,
+                                {false},
+                                {false}
+                              });
+
```

Comment:
```suggestion
  static hal::lpc40::spi spi2(2);
```

This needs to happen in the driver. Having the user remember to set the clock speed is putting too much responsibility on the application developer.